src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+                        case "chrY":
+                            contig = 23;
+                            break;
+                        default:
+                            try {
+                                contig = Integer.parseInt(splitLine[0].substring(3));
+                            } catch(NumberFormatException e) {
+                                continue loop_lines;
+                            }
+                    }
+                }
+                final int begin = Integer.parseInt(splitLine[1]) + 1;
+                final int end = Integer.parseInt(splitLine[2]) + 1;
+                final SVInterval svInterval = new SVInterval(contig, begin, end);
+                final SVIntervalTree.Entry<Double> entry = genome_intervals.find(svInterval);
+                // each interval has weight 1.0 for overlap. If multiple intervals exactly coincide, just add weight.
+                if(entry == null) {
+                    genome_intervals.put(svInterval, 1.0);
+                } else {
+                    entry.setValue(entry.getValue() + 1.0);
+                }
+            }
+        } catch(IOException e) {
+            throw new GATKException("Unable to load genome intervals from file \"" + genomeIntervalsFile + "\" line "
+                                    + lineNumber + ": " + e.getMessage());
+        }
+    }
+
+
+    @Override
+    public boolean hasNext() {
+        if ( listItr != null && listItr.hasNext() ) {
+            return true;
+        }
+        listItr = null;
+        boolean result = false;
+        while ( !result && treeItr.hasNext() ) {
+            final SVIntervalTree.Entry<List<BreakpointEvidence>> entry = treeItr.next();
+            final SVInterval curInterval = entry.getInterval();
+            final List<BreakpointEvidence> evidenceList = entry.getValue();
+            if( isValidated(entry.getValue()) || partitionCrossingChecker.onBoundary(curInterval) ) {
+                // already validated (no need to mark validated again) or on partition boundary (punt for now)
+                result = true;
+            } else if( anyPassesFilter(evidenceList) ) {
+                evidenceList.forEach(ev -> ev.setValidated(true));
+                result = true;
+            }
+
+            if ( result ) {
+                listItr = entry.getValue().iterator();
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public BreakpointEvidence next() {
+        if ( !hasNext() ) {
+            throw new NoSuchElementException("No next element.");
+        }
+        return listItr.next();
+    }
+
+    private boolean isValidated( final List<BreakpointEvidence> evList ) {
+        for ( final BreakpointEvidence ev : evList ) {
+            if ( ev.isValidated() ) return true;
+        }
+        return false;
+    }
+
+    boolean anyPassesFilter(final List<BreakpointEvidence> evidenceList) {
+        for(final BreakpointEvidence evidence : evidenceList) {
```

Comment:
I'd prefer to pass on this one. Basically every piece of evidence executes a loop with this code. Quick checks with JProfiler indicate adding around 2/3 cpu hours added with the stream version. Not huge but I think unnecessary. Ditto for isValidated().

Reply:
I'm surprised at your result here, @davidbenjamin 's code looks like it should be almost exactly equivalent to the for loop to me. Does your comment mean that it got 66% slower with the stream?

Reply:
There's just a tiny overhead, but each piece of evidence starts a for loop (or starts a stream). I profiled a tiny test case, got a roughly 0.1ms cost per BreakpointEvidence, which extrapolated to 40 cpu minutes over the course of a normal BAM file (of course we're running on a cluster, so wall clock increase is greatly less). Because the for loop is so fast, the proportional increase in anyPassesFilter is huge (several-fold), but that doesn't matter so much because it isn't a hot spot. The proportional increase in runtime of the whole pipeline is quite small (IMO just not worth the cost since the for-each loop is quite easy to understand).